core/src/main/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilder.java

```diff
@@ -81,12 +123,15 @@ public FunctionScoreQueryBuilder(ScoreFunctionBuilder scoreFunctionBuilder) {
      * @param filter the filter that defines which documents the function_score query will be executed on.
      * @param scoreFunctionBuilder score function that is executed
      */
-    public FunctionScoreQueryBuilder add(QueryBuilder filter, ScoreFunctionBuilder scoreFunctionBuilder) {
+    public FunctionScoreQueryBuilder add(QueryBuilder<?> filter, ScoreFunctionBuilder scoreFunctionBuilder) {
```

Comment:
I think we should get rid of the add and only allow to set entire lists or maybe go further an only allow one filter that needs to be wrapped in a BooleanQuery?